Sources/PlaydateKit/Core/Sound.swift

```diff
@@ -149,9 +149,139 @@ public enum Sound {
         private let pointer: OpaquePointer
     }
 
+    public class SamplePlayer {
+        // MARK: Lifecycle
+
+        /// Creates a new SamplePlayer.
+        public init() {
+            playerPointer = sampleplayer.newPlayer.unsafelyUnwrapped().unsafelyUnwrapped
+        }
+
+        deinit {
+            if samplePointer != nil {
+                sample.freeSample(samplePointer)
+            }
+            sampleplayer.freePlayer.unsafelyUnwrapped(playerPointer)
+        }
+
+        // MARK: Public
+
+        /// Assigns sample to player.
+        @discardableResult public func setSample(path: StaticString) -> Bool {
+            samplePointer = sample.load(path.utf8Start)
+            guard samplePointer != nil else { return false }
+
+            sampleplayer.setSample(playerPointer, samplePointer)
+            return true
+        }
+
+        /// Assigns sample to player.
+        @discardableResult public func setSample(path: UnsafePointer<CChar>) -> Bool {
+            samplePointer = sample.load(path)
+            guard samplePointer != nil else { return false }
+
+            sampleplayer.setSample(playerPointer, samplePointer)
+            return true
+        }
```

Comment:
I think you could probably call these methods more than once on the same SamplePlayer, resulting in the first samplePointer leaking. There should probably be a `if let samplePointer { sample.freeSample(samplePointer) }` at the start of both of them.

Reply:
Correct!